subsys/bluetooth/mesh/net.c

```diff
@@ -584,7 +584,6 @@ int bt_mesh_net_send(struct bt_mesh_net_tx *tx, struct net_buf *buf,
 	    BT_MESH_ADDR_IS_UNICAST(tx->ctx->addr)) {
 
 		err = 0;
-		goto done;
```

Comment:
Doesn't seem the condition has sense at all after removing `goto` since err is zero in any case.
Shouldn't it be instead:
```
if (IS_ENABLED(CONFIG_BT_MESH_GATT_PROXY) && 
    BT_MESH_ADDR_IS_UNICAST(tx->ctx->addr)) {
    (void)bt_mesh_proxy_relay(buf, tx->ctx->addr)
}
```

Reply:
Yes makes more sence. IMO it is better to avoid side effects in the if clause if it is possible, which it seems it is here.

Reply:
Please note that with `BT_MESH_ADDR_IS_UNICAST(tx->ctx->addr)` only unicast addresses will be relayed over GATT bearer, while group addresses should be relayed as well.

Reply:
@PavelVPV the unicast address check happens after proxy relaying is enqued. Now, that check has become redundant and can be removed.

Reply:
@omkar3141 this was a comment for the suggestion.